src/lib.rs

```diff
@@ -251,22 +250,19 @@ impl FailPoint {
                 Err(e) => panic!("unexpected poison: {:?}", e),
             }
 
-            let mut guard = self.pause.lock().unwrap();
-            *guard = false;
-            self.pause_notifier.notify_all();
+            self.pause.store(false, Ordering::Release);
         }
     }
 
+    #[cfg_attr(feature = "cargo-clippy", allow(clippy::option_option))]
     fn eval(&self, name: &str) -> Option<Option<String>> {
         let task = {
             let actions = self.actions.read().unwrap();
             match actions.iter().filter_map(|a| a.get_task()).next() {
                 Some(Task::Pause) => {
-                    let mut guard = self.pause.lock().unwrap();
-                    *guard = true;
+                    self.pause.store(true, Ordering::Release);
                     loop {
-                        guard = self.pause_notifier.wait(guard).unwrap();
-                        if !*guard {
+                        if !self.pause.load(Ordering::Acquire) {
```

Comment:
`pause` should make a thread sleep instead of busy loop.

>      pause   Threads sleep at the fail point until the fail point is set to
>	     off

https://www.freebsd.org/cgi/man.cgi?query=fail

Reply:
Oh, good catch. I'll revert this change.